server/database/models/UserRepository.js

```diff
@@ -0,0 +1,74 @@
+const AbstractRepository = require("./AbstractRepository");
+
+class UserRepository extends AbstractRepository {
+  constructor() {
+    super({ table: "user" });
+  }
+
+  async create(user) {
+    const [result] = await this.database.query(
+      `INSERT INTO ${this.table} (email, hashed_password, username, city) VALUES (?, ?, ?, ?)`,
+      [user.email, user.hashedPassword, user.username, user.city]
+    );
+
+    return result.insertId;
+  }
+
+  async readAll() {
+    const [rows] = await this.database.query(
+      `SELECT ${this.table}.*, p.image FROM ${this.table} LEFT JOIN picture as p ON p.user_id=${this.table}.id`
```

Comment:
By recovering all the information in this way, you also recover the password, creating a security hole. Correction in progress by Chris, so I'll let it slide this time.